File: cpp/TiketFilm.cpp
#include "TiketFilm.h"

// Constructor
TiketFilm::TiketFilm(
    string idTiket,
    string nomorKursi,
    int harga,
    string statusTiket,
    string judulFilm,
    string genre,
    int durasi,
    int ratingUsia
) : Tiket(idTiket, nomorKursi, harga, statusTiket) {

    this->judulFilm = judulFilm;
    this->genre = genre;
    this->durasi = durasi;
    this->ratingUsia = ratingUsia;
}

// Getter
string TiketFilm::getJudulFilm() {
    return judulFilm;
}

string TiketFilm::getGenre() {
    return genre;
}

int TiketFilm::getDurasi() {
    return durasi;
}

int TiketFilm::getRatingUsia() {
    return ratingUsia;
}

// Setter
void TiketFilm::setJudulFilm(string judulFilm) {
    this->judulFilm = judulFilm;
}

void TiketFilm::setGenre(string genre) {
    this->genre = genre;
}

void TiketFilm::setDurasi(int durasi) {
    this->durasi = durasi;
}

void TiketFilm::setRatingUsia(int ratingUsia) {
    this->ratingUsia = ratingUsia;
}

File: php/TiketFilm.php
<?php
require_once 'Tiket.php';

class TiketFilm extends Tiket
{
    private $judulFilm;
    private $genre;
    private $durasi;
    private $ratingUsia;
    private $gambar;

    public function __construct(
        $idTiket,
        $nomorKursi,
        $harga,
        $statusTiket,
        $judulFilm,
        $genre,
        $durasi,
        $ratingUsia,
        $gambar
    ) {
        parent::__construct($idTiket, $nomorKursi, $harga, $statusTiket);

        $this->judulFilm = $judulFilm;
        $this->genre = $genre;
        $this->durasi = (int) $durasi;
        $this->ratingUsia = (int) $ratingUsia;
        $this->gambar = $gambar;
    }

    public function getJudulFilm()
    {
        return $this->judulFilm;
    }

    public function getGenre()
    {
        return $this->genre;
    }

    public function getDurasi()
    {
        return $this->durasi;
    }

    public function getRatingUsia()
    {
        return $this->ratingUsia;
    }

    public function getGambar()
    {
        return $this->gambar;
    }

    public function setJudulFilm($judulFilm)
    {
        $this->judulFilm = $judulFilm;
    }

    public function setGenre($genre)
    {
        $this->genre = $genre;
    }

    public function setDurasi($durasi)
    {
        $this->durasi = (int) $durasi;
    }

    public function setRatingUsia($ratingUsia)
    {
        $this->ratingUsia = (int) $ratingUsia;
    }

    public function setGambar($gambar)
    {
        $this->gambar = $gambar;
    }
}
?>